entrypoint.sh

```diff
@@ -94,11 +94,20 @@ CommitChanges () {
     # Add only the destination directory
     git add --force "$DESTINATIONDIR"
     
-    # Commit all the changed to the the GitHub Pages branch
-    git commit -m "Auto commit" || exit 1
-    
-    # Push the changes to the remote GitHub Pages branch
-    git push
+    #check if there are any changes that are staged but not committed
+    if (! git diff --cached --exit-code --shortstat) 
+    then
+   	
+	# Commit all the changed to the the GitHub Pages branch
+    	git commit -m "Auto commit" || exit 1
+   	
+	# Push the changes to the remote GitHub Pages branch
+    	git push
+    else
+    	echo "There is no change in the documentation"
+    	exit 0
+    fi
+ 
```

Comment:
As suggested I added if(!git diff --cached --exit-code) to check if there are any changes that are staged but not committed before committing and pushing them.

I have also:

- used the option --shortstat to avoid the wall of text in the log created by git diff
- Added a warning message that inform the user that there is nothing to commit and then exit 0

I tested the code and it works well

Reply:
That sounds great! Thanks for the fix. I will remove the `exit 0` later because then `CommitChanges` function will have a side effect and no other function can be executed after the function call because it will exit from the whole shell script with 0 or 1 in any case (it is not a problem for now because it is the last function in the script but it is not future proof).